src/app/swap/swap-listener.ts

```diff
@@ -0,0 +1,68 @@
+import { toSats } from "@domain/bitcoin"
+
+import { admin as LedgerAdmin } from "@services/ledger/admin"
+import { WalletCurrency } from "@domain/shared"
+import { LedgerTransactionType } from "@domain/ledger"
+import { SwapProvider, SwapState, SwapType } from "@domain/swap"
+import { addAttributesToCurrentSpan } from "@services/tracing"
+import { SwapErrorNoActiveLoopdNode } from "@domain/swap/errors"
+import { sha256 } from "@domain/bitcoin/lightning"
+
+export const startSwapMonitor = async (swapService: ISwapService) => {
+  const isSwapServerUp = await swapService.healthCheck()
+  if (isSwapServerUp instanceof Error) return new SwapErrorNoActiveLoopdNode()
+  const listener = swapService.swapListener()
+  listener.on("data", (response) => {
+    handleSwapOutCompleted(response)
+  })
+}
+
+export const handleSwapOutCompleted = async (swapStatus: SwapStatusResultWrapper) => {
+  if (swapStatus.parsedSwapData) {
+    const type = swapStatus.parsedSwapData.swapType
+    const onchainMinerFee = toSats(swapStatus.parsedSwapData.onchainMinerFee)
+    const offchainRoutingFee = toSats(swapStatus.parsedSwapData.offchainRoutingFee)
+    const serviceProviderFee = toSats(swapStatus.parsedSwapData.serviceProviderFee)
+    const totalFees = onchainMinerFee + offchainRoutingFee + serviceProviderFee
```

Comment:
A few comments:
- type translations should be handled in the server (aka adapter). Ie `SwapStatusResultWrapper` should already have the correct types.
- you are using the 'old' `Satoshis` type - that type is deprecated. You should be using `BtcPaymentAmount` (right @vindard).